### PR TITLE
Fix .NET 8 build errors

### PR DESCRIPTION
**Pull request by jonathanpeppers**
Context: https://github.com/dotnet/maui/pull/12885/files

The main breaking change is that `MessagingCenter` is now `internal` / in the process of being removed.

Migrate to the messenger in the Community Toolkit instead.

I also updated the `NuGet.config` for the dotnet8 feed.

### CODE REVIEW

**Comment by jonathanpeppers** [comment]
This is working for me with .NET 8 Preview 4, but since this is using nightly builds, it is blocked by: https://github.com/dotnet/sdk/issues/32539

**Review by LoopedBard3 — APPROVED** [review]
LGTM, once in I will test adding it back for performance runs.

**Comment by jonathanpeppers** [comment]
We can try this again when this is merged & a new build is available:

https://github.com/dotnet/maui/pull/14997

**Pull request merged**
